### ISSUE TITLE
Support for Grafana Faro

### ISSUE THREAD

**Issue by paambaati** [uptrace/uptrace#201]
Support for frontend o11y/RUM via Grafana Faro would be nice.

From what I could tell, it uses OpenTelemetry, so perhaps it could work out of the box with Uptrace?

Ref —

1. https://grafana.com/oss/faro/
2. https://blog.mayflower.de/15107-grafana-faro-frontend-observability.html

**Comment by vmihailenco** [comment]
>From what I could tell, it uses OpenTelemetry, so perhaps it could work out of the box with Uptrace?

Yes, tracing should just work with Uptrace, but it is not clear if Faro is any better than opentelemetry-js, which is not very good when it comes to web/browsers integrations. That is why we try to support Sentry: https://uptrace.dev/get/ingest/sentry.html

Looks like it also use Prometheus remote write API and Loki, which are not supported right now but we have plans to support those APIs too.

Otel Collector might be helpful too: https://github.com/open-telemetry/opentelemetry-collector-contrib/issues/14751

**Comment by vmihailenco** [comment]
It does not look like Faro got a wide adoption and opentelemetry-js is slowly catching up so we'll probably stick with it for the immediate future.
